src/LaunchDarkly.EventSource/AsyncHelpers.cs

```diff
@@ -20,31 +20,35 @@ internal static class AsyncHelpers
             Func<CancellationToken, Task<T>> taskFn
             )
         {
-            var timeoutCancellation = new CancellationTokenSource(timeout);
-            var combinedCancellation = CancellationTokenSource.CreateLinkedTokenSource(
-                cancellationToken, timeoutCancellation.Token);
-            Task<T> task = null;
-            try
+            using (var timeoutCancellation = new CancellationTokenSource(timeout))
             {
-                task = taskFn(combinedCancellation.Token);
-                var result = await task;
-                return result;
-            }
-            catch (AggregateException e) when (e.InnerException is OperationCanceledException)
-            {
-                if (cancellationToken.IsCancellationRequested)
-                {
-                    throw e.InnerException;
-                }
-                throw new ReadTimeoutException();
-            }
-            catch (OperationCanceledException)
-            {
-                if (cancellationToken.IsCancellationRequested)
+                using (var combinedCancellation = CancellationTokenSource.CreateLinkedTokenSource(
+                    cancellationToken, timeoutCancellation.Token))
```

Comment:
The indentation change makes this diff hard to read, but the change here is simply that both `timeoutCancellation` and `combinedCancellation` are now scoped in `using` blocks so they'll be `Dispose`d in LIFO order no matter how we leave this method. Based on online discussions I've seen it seems that `CreateLinkedTokenSource` is more of a leak risk than `new CancellationTokenSource(TimeSpan)`, but there's no reason _not_ to dispose of them both of explicitly.